examples/bpk-component-modal/stories.tsx

```diff
@@ -0,0 +1,109 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2016 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import BpkModal from '../../packages/bpk-component-modal/src/BpkModal';
+
+import {
+  DefaultExample,
+  WideExample,
+  OverflowingExample,
+  CloseButtonTextExample,
+  LongTitleExample,
+  NotFullScreenOnMobileExample,
+  FullScreenExample,
+  FullScreenOverflowingExample,
+  NestedExample,
+  NoHeaderExample,
+  NoPaddingExample,
+  WithAccessoryViewExample,
+  ContrastExample,
+} from './examples';
+
+export default {
+  title: 'bpk-component-modal',
+  component: BpkModal,
+};
+
+export const Default = DefaultExample;
+export const Wide = WideExample;
+export const Overflowing = OverflowingExample;
+export const CloseButtonText = CloseButtonTextExample;
+
+export const LongTitle = LongTitleExample;
+
+export const NotFullScreenOnMobile = NotFullScreenOnMobileExample;
+
+export const FullScreen = FullScreenExample;
+
+export const FullScreenOverflowing = FullScreenOverflowingExample;
+
+export const Nested = NestedExample;
+export const NoHeader = NoHeaderExample;
+
+export const NoPadding = NoPaddingExample;
+
+export const WithAccessoryView = WithAccessoryViewExample;
+export const Contrast = ContrastExample;
+
+// Due to how iframes work we can pass a local url to load the stories above.
+// Attempted to use a Custom Iframe component with a react portal and ref to
+// render components but it didn't have the desired effect.
+const visualWrapper = (id: string, zoomEnabled: boolean = false) => (
+  <div style={{ height: '640px', width: '100%' }}>
+    <iframe
+      title={`Embedded Storybook ${id}`}
+      src={`/iframe.html?id=${id}&viewMode=story&args=zoomEnabled:${zoomEnabled}`}
```

Comment:
For deployed storybook we will have to have a think how we can make this work. As locally its fine because the address is

`localhost:9001/${src}` but on the deployed version the path is `backpack.github.io/storybook` so by doing `/iframe` it looks for just `backpack.github.io` which produces the following:

![Screenshot 2024-05-06 at 20 15 54](https://github.com/Skyscanner/backpack/assets/8831547/a5ef7bce-afaf-49db-b49c-b306583e83b7)

Reply:
Unless we don't mind for this test because its purely for visual tests? If so we should add a banner or text above the first iFrame to make it clear?

Reply:
Hmm that is is a good point. 
The other option was to try use the react portal again in the iframe.. I suppose we might not care for visual test scenarios in the deployed storybook? 🤔

Reply:
I don't think we can manage to load the modal in the iframe with a portal. 🤔 Is there a flag that could be passed in that adds the subpath to storybook so that the url can generate with/without or we just update the tests to match the release version?